File: src/main/java/com/github/blakefernandes/pulsar/items/action/types/MessageAction.java
package com.github.blakefernandes.pulsar.items.action.types;

import lombok.AllArgsConstructor;
import me.lucko.helper.text3.Text;
import org.bukkit.entity.Player;

@AllArgsConstructor
public class MessageAction extends Action {
    private String message;

    @Override
    public void execute(Player player) {
        player.sendMessage(Text.colorize(message));
    }
}

File: src/main/kotlin/com/github/blakefernandes/pulsar/actions/impl/MessageAction.kt
package com.github.blakefernandes.pulsar.actions.impl

import com.github.blakefernandes.pulsar.actions.Action
import com.github.blakefernandes.pulsar.messaging.Message
import com.github.blakefernandes.pulsar.messaging.Message.Placeholder
import org.bukkit.entity.Player

class MessageAction(
    private val message: String,
): Action() {
    override fun execute(player: Player, vararg placeholders: Placeholder) {
        Message(message).send(player, *placeholders)
    }

    override fun execute(player: Player) {
        Message(message).send(player)
    }
}